pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -136,6 +159,79 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 	// Don't modify the informer's copy.
 	tr := original.DeepCopy()
 
+	// build creation and status update adapted from prow: https://github.com/kubernetes/test-infra/blob/8faa040989cf8552683dcdd4400505ca62a3c408/prow/cmd/build/controller.go#L207
+	// create a build?
+	var wantBuild bool
+
+	switch {
+	case tr.DeletionTimestamp == nil:
+		wantBuild = true
+	}
+
+	// Get related task for taskrun
+	t, err := c.PipelineClientSet.PipelineV1alpha1().Tasks(tr.Namespace).Get(
+		tr.Spec.TaskRef.Name, metav1.GetOptions{})
+
+	var haveBuild bool
+
+	b, err := c.getBuild(namespace, name)
+	switch {
+	case errors.IsNotFound(err):
+		// Do not have a build
+	case err != nil:
+		return fmt.Errorf("get build: %v", err)
+	case b.DeletionTimestamp == nil:
+		haveBuild = true
+	}
+
+	// Should we create or delete this build?
+	switch {
+	case !wantBuild:
+		if !haveBuild {
+			fmt.Printf("Deleted %s", key)
+			return nil
+		}
+		switch or := metav1.GetControllerOf(b); {
+		case or == nil, or.APIVersion != groupVersionKind.GroupVersion().String(), or.Kind != groupVersionKind.Kind:
+			return nil // Not controlled by this
+		}
+		fmt.Printf("Delete builds/%s", key)
+		if err = c.deleteBuild(namespace, name); err != nil {
+			return fmt.Errorf("delete build: %v", err)
+		}
+		return nil
+	case finalState(tr.Status.State):
+		fmt.Printf("Finished %s", key)
+		return nil
+	case wantBuild && t.Spec.BuildSpec == nil:
+		return fmt.Errorf("nil BuildSpec")
+	case wantBuild && !haveBuild:
+		if b, err = makeBuild(*t); err != nil {
+			return fmt.Errorf("make build: %v", err)
+		}
+		fmt.Printf("Create builds/%s", key)
+		if b, err = c.createBuild(namespace, b); err != nil {
+			return fmt.Errorf("create build: %v", err)
+		}
+	}
+
+	// Ensure taskrun status is correct
+	haveState := tr.Status.State
```

Comment:
the `ntr` is not used anywhere?
Can we just move lines 219:232 to a function, and use that in L237 in reconcile ?

Reply:
this is an error on my part, ntr should be used in updateStatus.  I will need to move some things around for this

Reply:
I will refactor lines 219:232 to a function